docs/architecture/WritingSelectors.md

```diff
@@ -70,6 +70,29 @@ Component Selector name: [ComponentName]Selector
 
 A selector is a function built on top of reselect, it takes a list of functions as parameter and memoize their return value to decide if the second function provided in parameter need to be re-calculated, there are several principal we can follow when writing a selector for getting better perf:
 
+### 1. Parameters for select must be as minimun as possible
+
+The first array parameter of createSelector() defines an parameters array for memoizing the return value of selector. No extra data in parameters should be memoize when designing what parameters to be memoized,
+
+Don't:
+```typescript
+// state = {call:{ callId: UserId, displayname: string, callParticipants: Participants }}
+const getCall = (state) = > state.call;
+const callInfoSelector = createSelector([getCall], (call) => ({id: call.id, name: call.displayname }));
+```
+
+In this _Don't_ example, the memoize dependencies array includes the whole call object, which will trigger a lot of undesired re-rendering when new participant join(which will update whole call object), and this is causing a perf regression.
```

Comment:
nits:
```suggestion
In this _Don't_ example, the memoize dependency array includes the whole call object. This will trigger a lot of undesired re-rendering. For example, every time new participants join this will trigger an undesired re-render as when a participant joins the whole call object is updated.
```